2_process/src/create_obs_lake_task_plan.R

```diff
@@ -0,0 +1,128 @@
+
+do_obs_lake_tasks <- function(target_name, task_df_fn, irr_df_fn, k0_df_fn, ...){
+  
+  tasks <- tibble(task_filepath = names(yaml::yaml.load_file(task_df_fn))) %>% 
+    mutate(filename = basename(task_filepath)) %>% 
+    extract(filename, 'site_id', "split_obs_data_(.*).feather", remove = FALSE) %>% 
+    select(task_filepath, site_id) %>% 
+    filter(nchar(task_filepath) != 0) 
+  
+  irr_files <- tibble(filepath = names(yaml::yaml.load_file(irr_df_fn))) %>% 
+    extract(filepath, into = 'site_id', regex = "pb0_(.*)_irradiance.csv", remove = FALSE) %>% 
+    select(site_id, irr_filepath = filepath)
+  clarity_files <- tibble(filepath = names(yaml::yaml.load_file(k0_df_fn))) %>% 
+    extract(filepath, into = 'site_id', regex = "gam_(.*)_clarity.csv", remove = FALSE) %>% 
+    select(site_id, clarity_filepath = filepath)
+  
+  # Merge clarity and irr file names to use & remove sites that don't have both irr and kd
+  tasks <- tasks %>% 
+    left_join(irr_files, by = 'site_id') %>% 
+    left_join(clarity_files, by = 'site_id') %>%
+    filter(!is.na(irr_filepath) & !is.na(clarity_filepath))
+  
+  # ---- create task steps ---- #
+  
+  #' I'm doing this because each toha task is slow running
+  #' and if `morphometry` changes, we don't want to have to re-run all of them,
+  #' just the ones where _their_ morphometry changed. So we are subsetting first. 
```

Comment:
👍

Reply:
(full disclosure, credit goes to @jread-usgs who wrote that in one of our other task plans, but I wholeheartedly agree with it and reused the same approach here!)